src/umpire/DeviceAllocator.cpp

```diff
@@ -6,40 +6,63 @@
 //////////////////////////////////////////////////////////////////////////////
 #include "umpire/DeviceAllocator.hpp"
 
+#include <stdio.h>
+#include <string.h>
+
 #include "umpire/ResourceManager.hpp"
+#include "umpire/device_allocator_helper.hpp"
 #include "umpire/resource/MemoryResourceTypes.hpp"
 #include "umpire/util/Macros.hpp"
 
 namespace umpire {
 
-__host__ DeviceAllocator::DeviceAllocator(Allocator allocator, size_t size)
-    : m_allocator(allocator), m_ptr(static_cast<char*>(m_allocator.allocate(size))), m_size(size), m_child(false)
+__host__ DeviceAllocator::DeviceAllocator(Allocator allocator, size_t size, const char* name, size_t id)
```

Comment:
I would prefer if we could use std::string in the API and handle the conversion to char* internally.